gcc/d/dfrontend/expression.c

```diff
@@ -4082,6 +4082,12 @@ Expression *ArrayLiteralExp::semantic(Scope *sc)
         return new ErrorExp();
     }
 
+    if (global.params.noTypeinfo && elements->dim != 0)
+    {
+        error (global.params.noTypeinfo, "Can't use array literal");
```

Comment:
Just tried to make this a visitor like nogc. However,  when the visitor code runs `Type::getTypeInfo` has already been called. If I return an `ErrorExp` from `getTypeInfo` the visitor only gets an `ErrorExp` and I can't give proper error messages. Is there any good solution for this?